server/routing/lobby_test.go

```diff
@@ -196,3 +196,60 @@ func TestBadLobbyRequest(t *testing.T) {
 	}
 	assert.Contains(t, string(resp), "no lobby found with ID 'no-such-lobby'")
 }
+
+func TestLeavingLobby(t *testing.T) {
+	port, svcs := startTestServer()
+
+	// Create our lobby
+	lobbyIDBytes, err := post(port, LobbyRoute, []byte{})
+	if !assert.Nil(t, err) {
+		t.Fatal(err)
+	}
+
+	lobbyID := string(lobbyIDBytes)
+	playerName := "playerJuan"
+	leaveEndpoint := fmt.Sprintf("%v/%v/leave", LobbyRoute, lobbyID)
+
+	joinMsg := pogo.LobbyJoinMsg{
+		LobbyID:  lobbyID,
+		PlayerID: playerName,
+	}
+
+	_, err = put(port, LobbyRoute+"/join", joinMsg)
+	if !assert.Nil(t, err) {
+		t.Fatal(err)
+	}
+
+	lobbies := svcs.Lobby.GetLobbiesSnapshot()
+	if !assert.Len(t, lobbies[lobbyID].Players, 1) {
+		t.Fatal("expected 1 players in game lobby")
+	}
+	assert.Equal(t, playerName, lobbies[lobbyID].Players[0])
+
+	// Leave our lobby
+	leaveMsg := pogo.LobbyLeaveMsg{
+		PlayerID: playerName,
+	}
+
+	_, err = put(port, leaveEndpoint, leaveMsg)
+	if !assert.Nil(t, err) {
+		t.Fatal(err)
+	}
+
+	lobbies = svcs.Lobby.GetLobbiesSnapshot()
+	if !assert.Len(t, lobbies[lobbyID].Players, 0) {
+		t.Fatal("expected 0 players in game lobby")
```

Comment:
When there are 0 players in a lobby, should the lobby kill itself?  I can't remember, are there separate steps to creating and then joining a lobby?  Or is that one combined step?  If it is separate, then I guess having 0 players in the lobby is how it starts when it is first created.  But if it is joined, then the default lobby always has at least 1 player in it.  So if we allow players to leave lobbies now, we should probably handle an empty lobby.

Reply:
Yeah, it's two steps right now to create and join. But it still probably makes sense to close the lobby after the last player LEAVES it.